### PR TITLE
Tune JB backend dev loop

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

[JB dev loop with preview env: from 25mins to 25sec - Watch Video](https://www.loom.com/share/79097d8ceb584051b6aa7872450a3c3d)

This PR introduces a new workspace layer to deliver JB plugin independent from JB backend. It reduces build times from 15mins to 5mins on changes in JB backend plugin, since it does not require rebuild of all variants of JB backend images.

First https://github.com/gitpod-io/gitpod/pull/12426 should be deployed by workspace and webapp teams and all users migrated to new clusters.

It also adds several dev time scripts allowing [hot deployment](https://github.com/gitpod-io/gitpod/blob/ak/tune_jb_build/components/ide/jetbrains/backend-plugin/README.md#hot-deployment), [hot swapping](https://github.com/gitpod-io/gitpod/blob/ak/tune_jb_build/components/ide/jetbrains/backend-plugin/README.md#hot-swapping) and [remote debugging](https://github.com/gitpod-io/gitpod/blob/ak/tune_jb_build/components/ide/jetbrains/backend-plugin/README.md#remote-debugging). So one can directly adjust IDE configuration or a running workspace in preview environment without even committing changes.

gitpod-remote-hot drawio

## Related Issue(s)

Related https://github.com/gitpod-io/gitpod/issues/12268

## How to test

Start a workspace with:
- IntellIJ IDEA stable and latest as default IDE from dashboard
- GoLand stable and latest from JB Gateway while VS Code Browser is default IDE

#### How to verify that plugin version correspond to backend version (stable or latest)?
Screenshot 2022-08-26 at 11 33 56

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by akosyakov** [comment]
/hold 

First https://github.com/gitpod-io/gitpod/pull/12426 should be deployed by workspace and webapp teams and all users migrated to new clusters.

File: components/ide/jetbrains/backend-plugin/README.md
Comment (filiptronicek) [inline components/ide/jetbrains/backend-plugin/README.md:67]: suggestion(minor): spotted this one in the Loom you shared 😄 

```suggestion
Run `./remote-debug.sh <workspaceURL> (<localPort>)?` to configure remote debugging in a workspace running in preview environment.
```

**Review by sagor999 — APPROVED** [review]
Reviewed workspace-team related changes. LGTM

**Review by easyCZ — APPROVED** [review]
LGTM from WebApp side.

/hold in case you want another review

Worth noting that depending on the rollout ordering, we cannot guarantee that the server changes will be shipped before the plugin changes so some rollout management is needed from your side.

**Review by andreafalzetti — APPROVED** [review]

**Pull request merged**
